File: CPP/ByTier/B/1999_B_Card_Game.cpp
// Problem: CF 1999 B - Card Game
// https://codeforces.com/contest/1999/problem/B

/*
B. Card Game
	time limit per test2 seconds
	memory limit per test256 megabytes

Algorithms/Techniques: 
    - Enumeration of all possible game outcomes (4 combinations for 2 turns)
    - Comparison function using signum-like logic to determine round winner
    - Mathematical counting of favorable cases based on round results

Time Complexity: O(1) - Fixed number of operations per test case (4 combinations)
Space Complexity: O(1) - Only a constant amount of extra space used

Suneet and Slavic play a card game. The rules of the game are as follows:
Each card has an integer value between 1 and 10.
Each player receives 2 cards which are face-down (so a player doesn't know their cards).
The game is turn-based and consists exactly of two turns. In a round, both players pick a random unflipped card and flip it. The player who flipped a card with a strictly greater number wins the round. In case of equality, no one wins the round.
A player wins a game if he wins the most number of rounds (i.e. strictly greater than the other player). In case of equality, no one wins the game.
Since Suneet and Slavic aren't best friends, you need to calculate the number of ways the game could happen that Suneet would end up as the winner.
*/

#include <cstdio>
int f(int x, int y) { return (x > y) - (x < y); } // Signum function: returns 1 if x>y, -1 if x<y, 0 if x==y
int main() {
  int t;
  scanf("%d", &t);
  while (t--) {
    int a, b, c, d; // Cards of Suneet and Slavic: a,b for Suneet, c,d for Slavic
    scanf("%d %d %d %d", &a, &b, &c, &d);
    int res = 2 * (f(a, c) + f(b, d) > 0) + 2 * (f(a, d) + f(b, c) > 0); // Counting valid scenarios where Suneet wins
    printf("%d\n", res);
  }
}


// https://github.com/VaHiX/codeForces/

File: Kotlin/ByRound/1999/1999_B_Card_Game.kt
// Problem: CF 1999 B - Card Game
// https://codeforces.com/contest/1999/problem/B

/*
 * Problem: Card Game
 * 
 * Purpose: Calculate the number of ways Suneet can win a card game against Slavic,
 *          given that both players have two face-down cards each and the game consists
 *          of two turns where players flip one card each turn.
 * 
 * Approach:
 * - For each possible sequence of card flips (4 total combinations since each player
 *   has 2 cards), determine the outcome of both rounds.
 * - A player wins a round if their flipped card value is strictly greater than
 *   the opponent's.
 * - A player wins the game if they win more than half the rounds (i.e., 2 out of 2).
 * - Since we must enumerate all valid scenarios, we consider all combinations of
 *   which card each player flips in each round.
 * 
 * Time Complexity: O(1) - The number of test cases is bounded and the logic is constant.
 * Space Complexity: O(1) - Only constant extra space used.
 */

fun main() {
    val t = readLine()!!.toInt()
    for (test in 1..t) {
        val (a, b, c, d) = readLine()!!.split(' ').map { it.toInt() }
        var ways = 0
        // Check all valid combinations for Suneet to win the game
        // Case 1: Suneet flips a first, Slavic flips c first; Suneet flips b second, Slavic flips d second
        // If Suneet wins both rounds, increment ways by 2 (for a > c and b > d)
        if ((a > c && b >= d) || (a >= c && b > d))
            ways += 2
        // Case 2: Suneet flips a first, Slavic flips d first; Suneet flips b second, Slavic flips c second
        // If Suneet wins both rounds, increment ways by 2 (for a > d and b > c)
        if ((a > d && b >= c) || (a >= d && b > c))
            ways += 2
        print("$ways\n")
    }
}


// https://github.com/VaHiX/CodeForces/